extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -197,88 +197,97 @@ export class ResourceTypePickerDialog extends DialogBase {
 		const headerRowHeight = 28;
-		this._toolsTable.height = 25 * Math.max(toolRequirements.length, 1) + headerRowHeight;
+		this._toolsTable.height = 25 * Math.max(this.toolRequirements.length, 1) + headerRowHeight;
 		this._dialogObject.message = {
 			text: ''
 		};
 		this._installToolButton.hidden = true;
-		if (toolRequirements.length === 0) {
+		if (this.toolRequirements.length === 0) {
 			this._dialogObject.okButton.enabled = true;
 			this._toolsTable.data = [[localize('deploymentDialog.NoRequiredTool', "No tools required"), '']];
 			this._tools = [];
 		} else {
-			this._tools = toolRequirements.map(toolReq => {
+			this._tools = this.toolRequirements.map(toolReq => {
 				return this.toolsService.getToolByName(toolReq.name)!;
 			});
 			this._toolsLoadingComponent.loading = true;
 			this._dialogObject.okButton.enabled = false;
-			Promise.all(
-				this._tools.map(tool => tool.loadInformation())
-			).then(async () => {
-				// If the local timestamp does not match the class level timestamp, it means user has changed options, ignore the results
-				if (this.toolRefreshTimestamp !== currentRefreshTimestamp) {
-					return;
-				}
-				let minVersionCheckFailed = false;
-				const toolsToAutoInstall: ITool[] = [];
-				let messages: string[] = [];
-				this._toolsTable.data = toolRequirements.map(toolRequirement => {
-					const tool = this.toolsService.getToolByName(toolRequirement.name)!;
-					// subscribe to onUpdateData event of the tool.
-					this._toDispose.push(tool.onDidUpdateData((t: ITool) => {
-						this.updateToolsDisplayTableData(t);
-					}));
-					if (tool.isNotInstalled) {
-						if (tool.autoInstallSupported) {
-							toolsToAutoInstall.push(tool);
-						} else {
-							messages.push(localize('deploymentDialog.ToolInformation', "'{0}' was not discovered and automated installation is not currently supported. Install '{0}' manually or ensure it is started and discoverable. Once done please restart Azure Data Studio. See [{1}] .", tool.displayName, tool.homePage));
-						}
-					} else if (tool.isInstalled && toolRequirement.version && !tool.isSameOrNewerThan(toolRequirement.version)) {
-						minVersionCheckFailed = true;
-						messages.push(localize('deploymentDialog.ToolDoesNotMeetVersionRequirement', "'{0}' [ {1} ] does not meet the minimum version requirement, please uninstall it and restart Azure Data Studio.", tool.displayName, tool.homePage));
-					}
-
-					return [tool.displayName, tool.description, tool.displayStatus, tool.fullVersion || '', toolRequirement.version || '', tool.installationPath || ''];
-				});
+			let toolsLoadingErrors: string[] = [];
+			Promise.all(this._tools.map(tool => tool.loadInformation().catch(() => {
+				toolsLoadingErrors.push(`${tool.displayName}:${tool.statusDescription!}`);
+			})))
+				.then(() => this.executeToolsTableWorkflow(currentRefreshTimestamp, toolsLoadingErrors))
+				.catch(() => {/* Empty catch to keep linter happy */ });
```

Comment:
>/* Empty catch to keep linter happy */ [](start = 18, length = 38)

 Is there a reason you wouldn't at least console.log this? In the unlikely event that an error occurs it'd be good to have something to go on